TLM/TLM/Manager/Impl/ExtLaneManager.cs

```diff
@@ -0,0 +1,133 @@
+using CSUtil.Commons;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using TrafficManager.API.Manager;
+using TrafficManager.Patch;
+using TrafficManager.Util.Extensions;
+
+namespace TrafficManager.Manager.Impl {
+    internal class ExtLaneManager : AbstractCustomManager, IExtLaneManager {
+
+        private static readonly object lockObject = new object();
+
+        private readonly ExtLane[] lanes;
+
+        static ExtLaneManager() {
+            Instance = new ExtLaneManager();
+        }
```

Comment:
> Also, do we really need to instantiate so early?

Actually, it's better to use it that way -> setting property manually "throws away" implicit `if` check on access, see your example (not to mention that `if` condition looks really weird):

![image](https://user-images.githubusercontent.com/19638970/162578560-393b2b8c-840b-431b-96b0-956f944c44d4.png)

Compiles to this:

![image](https://user-images.githubusercontent.com/19638970/162578552-534b400c-8f1f-4855-9131-68cfca576005.png)


I also still don't see any reason for try caching contents of static constructor. It's job for the programmer to ensure that exception is impossible. using `try..catch` is not the best idea since it makes programmer to think less, and question still remains: what you do when you catch exception in static constructor? If it has thrown the exception then something is not right, shouldn't be used and also that code should never be merged into `master`. If you put something important in static constructor and you are not sure if can throw exception then it means you should rethink if the approach to solve that problem is right.
Maybe I'm weird but I never write code that throws exceptions to handle expected things 🤷‍♂️

Reply:
In general, catching exceptions that you can't report is a bad practice. Can we reliably report an exception in a static constructor?

Reply:
It must be shown to the user -> triggering UI dialog also may throw exception 😂

Reply:
The game won't report any exceptions in static constructor - it just fails horribly.

```csharp
try {
  // ...
} catch (Exception e) {
    e.LogException(true); // Shortcuts.cs
}
```

That should show error to user (`UIView.ForwardException()`), but no idea if that works from static constructor.